overlord/assertstate/validation_set_tracking.go

```diff
@@ -115,9 +133,12 @@ func ForgetValidationSet(st *state.State, accountID, name string) error {
 	if len(vsmap) == 0 {
 		return nil
 	}
+	if err := isForgetAllowedByModelAssertion(st, accountID, name); err != nil {
+		return err
```

Comment:
doesn't this path needs a test?

Reply:
It has tests in assertstate_test.go. There is a comment inside validation_set_tracking.go mentioning more extensive testing of ForgetValidationSet in assertstate_test.go. To keep this pattern I put them inside assertstate_test.go with the other tests of ForgetValidationSet